realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -50,15 +53,20 @@
 import io.realm.log.RealmLog;
 
 /**
- * This class represents the connection to the Realm Object Server for one {@link SyncConfiguration}.
+ * A session controls how data is synchronized between a single Realm on the device and the server
+ * Realm on the Realm Object Server.
  * <p>
- * A Session is created by opening a Realm instance using that configuration. Once a session has been created,
+ * A Session is created by opening a Realm instance using a {@link SyncConfiguration}. Once a session has been created,
  * it will continue to exist until the app is closed or all threads using this {@link SyncConfiguration} closes their respective {@link Realm}s.
  * <p>
- * A session is fully controlled by Realm, but can provide additional information in case of errors.
- * It is passed along in all {@link SyncSession.ErrorHandler}s.
+ * A session is controlled by Realm, but can provide additional information in case of errors.
+ * These errors are passed along in the {@link SyncSession.ErrorHandler}.
+ * <p>
+ * When creating a session, Realm will establish a connection to the server. This connection is
+ * controlled by Realm and might be shared between multiple sessions. It is possible to get insight
+ * into the connection using {@link #addConnectionChangeListener(ConnectionListener)} and {@link #isConnected()}.
```

Comment:
not sure if we should mention the possibility to get the session's state?

Reply:
I added a section below which explains this.